### PR TITLE
SS-10469 - Create coding patterns For Standard Table Component

### PR DESCRIPTION
**Pull request by ryanFurnessComprsa**
Accessibility updates for column sorting

### CODE REVIEW

File: src/browser/sorterRow.html [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-  -->
+<tr class="ng-table-sort-header" role="row">  
   <!-- Checkbox template must be in a <td> as per SS-20165 -->
-  <td ng-repeat="$column in $columns" ng-init="template = $column.headerTemplateURL(this)" ng-if="$column.headerTemplateURL(this)">
+  <td ng-repeat="$column in $columns" ng-init="template = $column.headerTemplateURL(this)" ng-if="$column.show(this) && $column.headerTemplateURL(this)">
     <div ng-if="template" ng-include="template"></div>
   </td>
-  <th scope="col" title="{{$column.headerTitle(this)}}" role="columnheader" ng-attr-aria-sort="{{!$column.sortable(this) ? undefined : params.sorting()[$column.sortable(this)]=='asc' ? 'ascending' : params.sorting()[$column.sortable(this)]=='desc' ? 'descending' : 'none'}}"
-    ng-repeat="$column in $columns" ng-class="{ 
+  <th scope="col" title="{{$column.headerTitle(this)}}" role="columnheader" ng-attr-aria-sort="{{!$column.sortable(this) ? 'none' : params.sorting()[$column.sortable(this)]=='asc' ? 'ascending' : params.sorting()[$column.sortable(this)]=='desc' ? 'descending' : 'none'}}"
```
Comment (ajayvjain) [inline src/browser/sorterRow.html:6]: Can we translate these strings?

File: src/core/ngTableParams.ts [hunk 2]
```diff
@@ -697,7 +697,10 @@ export function ngTableParamsFactory<T>(
                 more: 'More Pages', // default values
                 next: 'Next Page', // default values 
                 prev: 'Previous Page', // default values
-                pageNumPrefix: 'Page'
+                pageNumPrefix: 'Page', // default values
+                sortDescriptionAsc: 'sort in ascending order', // default values
```
Comment (ajayvjain) [inline src/core/ngTableParams.ts:701]: Can we translate these strings?

File: test/tableDynamicSpec.spec.ts [hunk 3]
```diff
@@ -638,7 +638,7 @@ describe('ng-table-dynamic', () => {
             var rows = thead.find('tr');
             var titles = ng1.element(rows[0]).find('th');
 
-            return ng1.element(titles).text().trim().split(/\s+/g)
+            return ng1.element(titles).text().trim().replace('sort in ascending order','').replace('sort in ascending order','').replace('sort in ascending order','').trim().split(/\s+/g)
```
Comment (ajayvjain) [inline test/tableDynamicSpec.spec.ts:641]: Can we translate these strings?

**Review by ajayvjain — COMMENTED** [review]
We need to support translations for providing sort instructions. Can you please check if that can be done? Or if we can pass translated strings to this component?

File: src/browser/sorterRow.html (hunk 1 above)
Comment (ryanFurnessComprsa) [reply src/browser/sorterRow.html:6]: The strings for the column headers can be parsed in, in translated state when the table is instantiated on the frontend. I think this is already being done in the system.

File: src/core/ngTableParams.ts (hunk 2 above)
Comment (ryanFurnessComprsa) [reply src/core/ngTableParams.ts:701]: Yes, in settings object there is a property called "translations" that when parsed with the correct properties replaces the property values in the accessibilityOptions object. This "translations" property can be parsed in when the table is instantiated on the frontend.

**Pull request merged**